kernel/src/hil/flash.rs

```diff
@@ -12,14 +16,14 @@ pub trait Flash {
     /// when operations complete.
     fn set_client(&self, client: &'static Client);
 
-    /// Read data
-    fn read(&self, offset: usize, buf: &'static mut [u8]);
+    /// Read a page of flash.
+    fn read_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
 
-    /// Write data
-    fn write(&self, offset: usize, buf: &'static mut [u8]);
+    /// Write a page of flash.
+    fn write_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
```

Comment:
Any implementation will have to truncate `buf` if it is larger than the page size, right? If so, perhaps it should be made explicit in the comment?